ledger_device_sdk/src/io_new.rs

```diff
@@ -0,0 +1,650 @@
+use crate::seph::{self, PacketTypes};
+
+pub use crate::io_legacy::{ApduHeader, Event, Reply, StatusWords};
+
+use crate::io_callbacks::nbgl_register_callbacks;
+
+#[cfg(any(
+    target_os = "nanox",
+    target_os = "stax",
+    target_os = "flex",
+    target_os = "apex_p"
+))]
+use crate::seph::ItcUxEvent;
+
+use ledger_secure_sdk_sys::seph as sys_seph;
+use ledger_secure_sdk_sys::*;
+
+#[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
+use crate::buttons::ButtonEvent;
+#[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
+use ledger_secure_sdk_sys::buttons::{get_button_event, ButtonsState};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CommError {
+    Overflow,
+    IoError,
+}
+
+pub const DEFAULT_BUF_SIZE: usize = 273;
+
+pub struct Comm<const N: usize = DEFAULT_BUF_SIZE> {
+    buf: [u8; N],
+    expected_cla: Option<u8>,
+
+    apdu_type: u8,
+    #[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
```

Comment:
To be replaced with `#[cfg(any(target_os = "nanosplus", target_os = "nanox"))]`

Reply:
None in 8d3bec3512e3678a926280f0b289402aae534277.

At some point we should add better aliases for the various device groups/features, so this could become something similar to

```
#[cfg(has_buttons)]
```